src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(state, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const keyToRemove of action.keysToRemove) {
+        delete state[keyToRemove];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (action in state) {
+        delete state[action];
+      }
```

Comment:
you are already have `action` variable
```suggestion
      for (key in state) {
        delete state[key];
      }
```